app/js/lib/foxbox/network.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+// Private members.
+const p = {
+  // Private properties.
+  // Whether we can connect to the box via a local connection.
+  local: Symbol('local'),
+  // Whether we can connect to the box via a remote connection.
+  remote: Symbol('remote'),
+  // A reference to the interval to get the online status.
+  pingInterval: Symbol('pingInterval'),
+
+  // Private methods.
+  fetch: Symbol('fetch'),
+  ping: Symbol('ping'),
+  pingBox: Symbol('pingBox')
+};
+
+export default class Network {
+  constructor(settings) {
+    this.settings = settings;
+
+    this[p.local] = false;
+    this[p.remote] = false;
+    this[p.pingInterval] = null;
+  }
+
+  /**
+   * Attach event listeners related to the connection status.
+   *
+   * @return {Promise}
+   */
+  init() {
+    const pingBox = this[p.pingBox].bind(this);
+
+    window.addEventListener('online', pingBox);
+    window.addEventListener('offline', pingBox);
+
+    if ('connection' in navigator && 'onchange' in navigator.connection) {
+      navigator.connection.addEventListener('change', pingBox);
+
+      // We also ping the box every few minutes to make sure it's still there.
+      this[p.pingInterval] = setInterval(pingBox,
+        this.settings.onlineCheckingLongInterval);
+    } else {
+      // If the Network Information API is not implemented, fallback to polling.
+      this[p.pingInterval] = setInterval(pingBox,
+        this.settings.onlineCheckingInterval);
+    }
+
+    this[p.pingBox]();
+
+    return Promise.resolve();
+  }
+
+  get origin() {
+    if (this[p.local]) {
+      return this.localOrigin;
+    } else if (this[p.remote]) {
+      return this.tunnelOrigin;
+    }
+
+    console.error('The box is out of reach.');
+    return this.localOrigin;
+  }
+
+  get localOrigin() {
+    return `${this.settings.localScheme}://${this.settings.localHostname}:${this.settings.localPort}`;
+  }
+
+  get tunnelOrigin() {
+    return `${this.settings.tunnelScheme}://${this.settings.tunnelHostname}:${this.settings.tunnelPort}`;
+  }
+
+  get connected() {
+    return this[p.local] || this[p.remote];
+  }
+
+  get connection() {
+    if (this[p.local]) {
+      return 'local';
+    } else if (this[p.remote]) {
+      return 'remote';
+    }
+    return 'unknown';
+  }
+
+  /**
+   * Request a JSON from a specified URL.
+   *
+   * @param {string} url The URL to send the request to.
+   * @param {string} method The HTTP method (defaults to "GET").
+   * @param {Object} body An object of key/value.
+   * @return {Promise}
+   */
+  fetchJSON(url, method = 'GET', body = undefined) {
+    return this[p.fetch](url, 'application/json', method, body)
+      .then(response => response.json());
```

Comment:
__Code Style Proposal:__ I'd propose always use ```() => {} and (a, b, c) => {}``` arrow function notation so that we always use the same style for all types of arrow functions and no matter if they have one, multiple or no parameters at all. IMO it's a bit easier to read, but it's subjective thing. What do you think?